src/telliot_core/contract/contract.py

```diff
@@ -107,15 +117,38 @@ async def write(
             contract_function = self.contract.get_function_by_name(func_name)
             transaction = contract_function(**kwargs)
 
-            built_tx = transaction.buildTransaction(
-                {
-                    "from": acc.address,
-                    "nonce": acc_nonce,
-                    "gas": gas_limit,
-                    "gasPrice": self.node.web3.toWei(gas_price, "gwei"),
-                    "chainId": self.node.chain_id,
-                }
-            )
+            # start tx dict with static elements
+            tx_dict = {
+                "from": acc.address,
+                "nonce": acc_nonce,
+                "gas": gas_limit,
+                "chainId": self.node.chain_id,
+            }
+
+            # if legacy gas price is not None, use it
+            if legacy_gas_price is not None:
+                tx_dict["gasPrice"] = self.node.web3.toWei(legacy_gas_price, "gwei")
+
+            # else if (if legacy gas price is None) and max fee is not None, use max fee
+            elif max_fee_per_gas is not None:
+                tx_dict["maxFeePerGas"] = self.node.web3.toWei(max_fee_per_gas, "gwei")
+
+                if max_priority_fee_per_gas is not None:
+                    tx_dict["maxPriorityFeePerGas"] = self.node.web3.toWei(
+                        max_priority_fee_per_gas, "gwei"
+                    )
+
+            # else if (if legacy price and max fee are not provided), use max priority fee
+            elif max_priority_fee_per_gas is not None:
+                tx_dict["maxPriorityFeePerGas"] = self.node.web3.toWei(
+                    max_priority_fee_per_gas, "gwei"
+                )
+
+            # else (if none are provided) estimate max priority fee with rpc node
+            else:
+                tx_dict["maxPriorityFeePerGas"] = self.node.web3.eth.max_priority_fee
```

Comment:
Have you verified that this puts out a good value (and the units)? 
Also, is this reachable given the ValueError raised above when there are no arguments?